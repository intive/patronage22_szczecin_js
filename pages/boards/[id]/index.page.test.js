/* globals describe, it, expect */

import { render, screen, waitFor } from '@testing-library/react'
import BoardDetails from './index.page'
import userEvent from '@testing-library/user-event'

describe('BoardDetails', () => {
  const tileDetails = {
    columns: [
      {
        id: '01FV73XQTDV2Q6YWMGG8Z4A0NE',
        name: 'column 1',
        cards: [
          {
            id: '01FV75BM8W3ZDZHSQE36AART7B',
            text: 'card 1'
          },
          {
            id: '01FV75BQYD91NGEGE9V2Q1ZTMY',
            text: 'card 2'
          },
          {
            id: '01FV75BTSWEZYV8786KJTY3KSS',
            text: 'card 3'
          }
        ]
      },
      {
        id: '01FV73XQTDV2Q6YWMGG8Z4A0N1',
        name: 'column 2',
        cards: [
          {
            id: '01FV75BM8W3ZDZHSQE36AART7V',
            text: 'card 1'
          }
        ]
      }
    ],
    name: 'board 3',
    id: '01FV75A884ZHZE347P83K0SXRR',
    password: false
  }

  it('should render BoardDetails', async () => {
    const { container } = render(<BoardDetails tileDetails={tileDetails} />)
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })

  it('should open Modal after button click and close it after click on backdrop', () => {
    render(<BoardDetails tileDetails={tileDetails} />)
    expect(screen.queryByTestId('backdrop')).not.toBeInTheDocument()
    userEvent.click(screen.queryByRole('button', { name: /boardHeader.buttonText/i }))
    expect(screen.getByTestId('backdrop')).toBeInTheDocument()
    userEvent.click(screen.getByTestId('backdrop'))
    expect(screen.queryByTestId('backdrop')).not.toBeInTheDocument()
  })

  it('should open Modal after button click and close it after click on cancel button', () => {
    render(<BoardDetails tileDetails={tileDetails} />)
    expect(screen.queryByRole('button', { name: /buttons.cancel/i })).not.toBeInTheDocument()
    userEvent.click(screen.queryByRole('button', { name: /boardHeader.buttonText/i }))
    expect(screen.getByRole('button', { name: /buttons.cancel/i })).toBeInTheDocument()
    userEvent.click(screen.getByRole('button', { name: /buttons.cancel/i }))
    expect(screen.queryByRole('button', { name: /buttons.cancel/i })).not.toBeInTheDocument()
  })

  it('should open Modal after button click and close it after click on Esc keyboard button', () => {
    render(<BoardDetails tileDetails={tileDetails} />)
    expect(screen.queryByRole('button', { name: /buttons.save/i })).not.toBeInTheDocument()
    userEvent.click(screen.queryByRole('button', { name: /boardHeader.buttonText/i }))
    expect(screen.getByRole('button', { name: /buttons.save/i })).toBeInTheDocument()
    userEvent.keyboard('{esc}')
    expect(screen.queryByRole('button', { name: /buttons.save/i })).not.toBeInTheDocument()
  })

  it('should open Modal after button click and close it after click on save button if column Name length is more then 5', () => {
    render(<BoardDetails tileDetails={tileDetails} />)
    expect(screen.queryByRole('button', { name: /buttons.save/i })).not.toBeInTheDocument()
    userEvent.click(screen.queryByRole('button', { name: /boardHeader.buttonText/i }))
    expect(screen.getByRole('button', { name: /buttons.save/i })).toBeInTheDocument()
    userEvent.type(screen.getByRole('textbox'), 'New Column')
    expect(screen.getByRole('textbox')).toHaveValue('New Column')
    expect(screen.queryByRole('button', { name: /buttons.save/i })).not.toBeDisabled()
    userEvent.click(screen.getByRole('button', { name: /buttons.save/i }))
    expect(screen.queryByRole('button', { name: /buttons.save/i })).not.toBeInTheDocument()
  })

  it('should open Modal after button click and save button to be disabled if password length less than 5', () => {
    render(<BoardDetails tileDetails={tileDetails} />)
    expect(screen.queryByRole('button', { name: /buttons.save/i })).not.toBeInTheDocument()
    userEvent.click(screen.queryByRole('button', { name: /boardHeader.buttonText/i }))
    expect(screen.getByRole('button', { name: /buttons.save/i })).toBeInTheDocument()
    userEvent.type(screen.getByRole('textbox'), '1234')
    expect(screen.getByRole('textbox')).toHaveValue('1234')
    expect(screen.queryByRole('button', { name: /buttons.save/i })).toBeDisabled()
    expect(screen.queryByRole('button', { name: /buttons.save/i })).toBeInTheDocument()
  })
})
