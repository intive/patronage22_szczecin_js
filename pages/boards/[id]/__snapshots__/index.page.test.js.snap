// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardDetails should render BoardDetails 1`] = `
<div>
  <header
    class="sc-furwcr gznTzn"
    data-testid="header"
  >
    <div
      class="sc-pVTFL doYdbH"
    >
      <a
        class="sc-jrQzAO HnibN"
        href="/"
      >
        <span
          class="material-icons"
        >
          crop_original
        </span>
        <p
          class="sc-kDTinF lpfmpE"
        >
          headerTitle
        </p>
      </a>
    </div>
  </header>
  <section
    class="sc-dkPtRN fQLMUS"
    data-testid="board-header"
  >
    <a
      class="sc-jRQBWg irLjuY"
      href="/"
    >
      boardHeader.returnLinkText
    </a>
    <div
      class="sc-hKwDye hETRbD"
    >
      <h1
        class="sc-eCImPb cIClSu"
      >
        board 3
      </h1>
      <button
        class="sc-bdvvtL dRBcfm"
      >
        boardHeader.buttonText
      </button>
    </div>
  </section>
  <div
    class="sc-hGPBjI eIsNRz"
  >
    <div
      class="sc-dJjYzT fMzkQP"
    >
      <div
        class="sc-iqseJM fsyZAX"
        data-testid="board-column"
      >
        <div
          class="sc-crHmcD fUwZby"
        >
          <h4
            class="sc-egiyK cEBQIG"
          >
            column 1
          </h4>
          <button
            class="sc-bdvvtL boJPqq sc-bqiRlB dRlpIO"
          >
            boardColumn.addCard
          </button>
        </div>
        <div>
          <div
            class="sc-crHmcD czukwB"
          >
            <div
              class="sc-ksdxgE emXTuD"
            >
              card 1
            </div>
            <div
              class="sc-hBUSln lkdYcG"
            >
              <span
                class="sc-fotOHu jKaTeu"
              />
              <span
                class="material-icons sc-fFeiMQ fedWoZ"
              >
                favorite_border
              </span>
            </div>
          </div>
          <div
            class="sc-crHmcD czukwB"
          >
            <div
              class="sc-ksdxgE emXTuD"
            >
              card 2
            </div>
            <div
              class="sc-hBUSln lkdYcG"
            >
              <span
                class="sc-fotOHu jKaTeu"
              />
              <span
                class="material-icons sc-fFeiMQ fedWoZ"
              >
                favorite_border
              </span>
            </div>
          </div>
          <div
            class="sc-crHmcD czukwB"
          >
            <div
              class="sc-ksdxgE emXTuD"
            >
              card 3
            </div>
            <div
              class="sc-hBUSln lkdYcG"
            >
              <span
                class="sc-fotOHu jKaTeu"
              />
              <span
                class="material-icons sc-fFeiMQ fedWoZ"
              >
                favorite_border
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-iqseJM fsyZAX"
        data-testid="board-column"
      >
        <div
          class="sc-crHmcD fUwZby"
        >
          <h4
            class="sc-egiyK cEBQIG"
          >
            column 2
          </h4>
          <button
            class="sc-bdvvtL boJPqq sc-bqiRlB dRlpIO"
          >
            boardColumn.addCard
          </button>
        </div>
        <div>
          <div
            class="sc-crHmcD czukwB"
          >
            <div
              class="sc-ksdxgE emXTuD"
            >
              card 1
            </div>
            <div
              class="sc-hBUSln lkdYcG"
            >
              <span
                class="sc-fotOHu jKaTeu"
              />
              <span
                class="material-icons sc-fFeiMQ fedWoZ"
              >
                favorite_border
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
